llvm/utils/TableGen/Common/CodeGenRegisters.cpp

```diff
@@ -411,6 +410,9 @@ CodeGenRegister::computeSubRegs(CodeGenRegBank &RegBank) {
         Parts.push_back(getSubRegIndex(SR->ExplicitSubRegs[j]));
     }
 
+    if (Parts.size() < 2)
+      continue;
```

Comment:
Shouldn't we `continue` here if _any_ of the subreg indices of SR are artificial? (Either instead of or as well as the `Parts.size()` condition.) E.g. if SR had _two_ real subreg indices and one artificial one, we should give up?

Reply:
Good point @jayfoad!

Reply:
@jayfoad are you happy with the latest change? (apologies for the prod, I wasn't sure if you noticed I had addressed your comment)